### ISSUE TITLE
Error when trying to start engine

### ISSUE THREAD

**Issue by cmilam87** [itchannel/fordpass-ha#278]
When trying to start the engine on my car I'm currently getting this error in the logs and the car does not start. Same thing happens when trying to stop. The un/lock works just fine. 

I'm guessing the response from the Ford API has changed as it seems to be looking for the `status` off of the `result` and it is not present.

Using version 1.47 of the integration.
Using version 1.32.1 of HACS
Using version 2023.7.3 of Home Assistant

```
2023-07-28 18:34:55.360 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [140085037622608] 'status'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 205, in handle_call_service
    await hass.services.async_call(
  File "/usr/src/homeassistant/homeassistant/core.py", line 1965, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2005, in _execute_service
    return await cast(
           ^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 235, in handle_service
    return await service.entity_service_call(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 848, in entity_service_call
    response_data = task.result()  # pop exception if have
                    ^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1192, in async_request_call
    return await coro
           ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 892, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/config/custom_components/fordpass/switch.py", line 43, in async_turn_on
    await self.coordinator.hass.async_add_executor_job(
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/fordpass/fordpass_new.py", line 383, in start
    return self.__requestAndPoll(
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/fordpass/fordpass_new.py", line 484, in __requestAndPoll
    return self.__pollStatus(url, result["commandId"])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/fordpass/fordpass_new.py", line 467, in __pollStatus
    if result["status"] == 552:
       ~~~~~~^^^^^^^^^^
```

**Comment by itchannel** [comment]
@cmilam87 I don't receive an error when testing on my vehicles. So must be specific to your instance.

Can you add the debug line above that if statement mentioned in your error log and then enable debugging so I can see what's being returned from the API. It sounds like your vehicle is replying with an error or the API was unable to perform that function. Do you have a low battery or remote start has been disabled?
```
_LOGGER.debug(result)
if result["status"] == 552:
```

**Comment by cmilam87** [comment]
Well that's odd. It seems to be working now.

I added the `_LOGGER.debug(result)` but I'm not seeing any output. I've got HACS debug logging enabled with this configuration

```
logger:
  default: info
  logs:
    custom_components.hacs: debug
    aiogithubapi: debug
```

Only thing I can think of is maybe it was in a deep sleep at the time and didn't respond with the proper status? I did not note down what the `fordpass_deepSleepInProgress` status was at the time unfortunately.

**Comment by cmilam87** [comment]
Well it happened again, but my dumb monkey brain saw an update for this plugin and went ahead and downloaded it without realizing it would overwrite the debug logging.

I think I'm onto something though with the car being in deep sleep, as it happened after the car had been off all night. I'll see if I can get it to recreate it now that the debug logging is back in and I'm on the latest version.

**Comment by cmilam87** [comment]
Okay - I think I've figured out what's causing the issue, but for some reason I'm still not seeing the debug logs.

It has to do with the Google Assistant integration. Whenever I tell Google to lock the door of the car it throws that error. However, if I use the switch on the Home Assistant dashboard it works just fine.

I'll do some more digging.

**Comment by cmilam87** [comment]
Looks like when I use the Google Assistant integration it is throwing a 404. I ended up just adding a info log and was able to get the result.

```
2023-07-30 13:02:39.161 INFO (SyncWorker_0) [custom_components.fordpass.fordpass_new] {'statusCode': 404, 'message': 'Resource not found'}
2023-07-30 13:02:39.162 ERROR (MainThread) [homeassistant.core] Error executing service: <ServiceCall switch.turn_on (c:01H6KW108JEWDJHWGYGHBGCBT8): entity_id=['switch.fordpass_ignition_switch']>
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/core.py", line 1983, in catch_exceptions
    await coro_or_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 2005, in _execute_service
    return await cast(
           ^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 235, in handle_service
    return await service.entity_service_call(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 848, in entity_service_call
    response_data = task.result()  # pop exception if have
                    ^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1192, in async_request_call
    return await coro
           ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 892, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/config/custom_components/fordpass/switch.py", line 43, in async_turn_on
    await self.coordinator.hass.async_add_executor_job(
  File "/usr/local/lib/python3.11/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/fordpass/fordpass_new.py", line 400, in start
    return self.__requestAndPoll(
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/fordpass/fordpass_new.py", line 504, in __requestAndPoll
    return self.__pollStatus(url, result["commandId"])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/fordpass/fordpass_new.py", line 486, in __pollStatus
    if result["status"] == 552:
       ~~~~~~^^^^^^^^^^
KeyError: 'status'
```

**Comment by cmilam87** [comment]
I figured out the issue and it's kind of funny and makes sense now. Because I named the doors and locks of my car in the possessive form (Jolene's Doors, Jolene's Engine) - it contained an apostrophe. I believe the URL is not getting properly encoded because as soon as I renamed it to Jolene Doors/Engine, the Google Assistant integration works just fine.

What I'm unsure of is if that is something from the Google Assistant integration, or something on this integration.
